keps/sig-architecture/2140-receipts/README.md

```diff
@@ -0,0 +1,464 @@
+
+The flows shown in the state diagram are described in the sections below. Some of the flows described will have appropriate `kepctl` commands from the `Alpha` version itself. Other helper commands will be added and existing commands refined throughout the rollout process.
+### Initializing the Receipts directory for a Release
+
+In the initial rollout of this process, the release team will manually generate this structure, however, `kepctl` will provide a command to generate this structure:
+
+
+```
+$ kepctl create release v1.22
+==> 1.22 Release Created
+$ tree releases
+releases
+└── v1.22
+    ├── accepted
+    ├── at-risk
+    ├── exception
+    ├── OWNERS
+    ├── proposed
+    └── removed
+```
+
+### Proposing a KEP for a Release
+
+A KEP author or SIG Chair/Tech Lead opts-in to a given release by creating the receipt document ([described above](#where-are-the-receipts-stored)) in the `proposed`  directory for the specified release. The `latest-milestone` and `stage` should be updated in the associated `kep.yaml` file as well. This process can be done manually or with `kepctl`.
+
+```
+$ kepctl promote sig-architecture/000-mykep --stage beta --release v1.22
+==> KEP sig-architecture/000-mykep updated
+==> 1.22 Release Receipt Created For sig-architecture/000-mykep
+==> Please Open a Pull Request to Finish Proposing This Enhancement for v1.22
+```
+
+A PR should then be opened by the KEP Author to complete the opt-in process. Once the PR is opened, a set of validation will run against the KEP based on the metadata in the receipt to ensure that all machine verifiable parts of the KEP are correctly populated for the specified stage. If required, when the validation completes, the enhancements team for that release will approve the PR. The KEP is now in the `proposed` state.
+
+The `promote` command will also check for PRR requirements when creating the Receipt. This will pro-actively inform the KEP Authors of any issues with the KEP.
+
+### Accepting a KEP for a Release
+
+Once a receipt has been merged to the `proposed` state, the enhancements team will do further review of the KEP to validate things that cannot currently be automatically validated. Each `proposed` KEP will then be moved to the `accepted` directory, if all requirements are met. If the KEP is currently missing any required elements for the specified stage, it will be moved to the `at-risk` directory. This will be done periodically by the enhancements team, and new PRs will be generated to move the receipts to the proper location(s).
+
+
+We will enhance `kepctl` to add an `accept` command that will perform any available validation and handle the movement of the `receipt`:
+
+```
+$ tree releases
+releases
+└── v1.22
+    ├── accepted
+    │   └── 0001.yaml
+    ├── at-risk
+    │   └── 0002.yaml
+    ├── exception
+    ├── OWNERS
+    ├── proposed
+    │   └── 0004.yaml
+    └── removed
+        └── 0003.yaml
+
+$ kepctl accept 0004 --at-risk --release v1.22
```

Comment:
we can use this after enhancements freeze to indicate when an enhancement is at risk of not making the code freeze. But, before enhancement freeze, I think I like John's idea of a simpler structure and open PR against the accepted directory. We can mark things at risk in the PR itself instead of keeping track here.